db/adapter.go

```diff
@@ -0,0 +1,30 @@
+package db
+
+import "errors"
+
+type readerRWAdapter struct{ DBReader }
+
+var (
+	// ErrReadOnly is returned when a write operation is attempted on a read-only transaction.
+	ErrReadOnly = errors.New("cannot modify read-only transaction")
+)
+
+// Returns a ReadWriter that forwards to a reader and errors if writes are
+// attempted. Can be used to pass a Reader when a ReadWriter is expected
+// but no writes will actually occur.
+func NewReadWriterFromReader(r DBReader) DBReadWriter {
```

Comment:
- Godoc doesn't match the function name.
- The name stutters quite a bit. Can we consider renaming this perhaps?

Reply:
maybe `ReaderAsReadWriter`? it's marginally shorter, but still clear

Reply:
Yeah, that's an improvement IMO. ACK 👍